pkg/networks/usernet/UDPFileConn.go

```diff
@@ -0,0 +1,21 @@
+package usernet
+
+import (
+	"errors"
+	"net"
+	"time"
+)
+
+type UDPFileConn struct {
+	net.Conn
+}
+
+func (conn *UDPFileConn) Read(b []byte) (n int, err error) {
+	// Check if the connection has been closed
+	if err := conn.SetReadDeadline(time.Time{}); err != nil {
+		if opErr, ok := err.(*net.OpError); ok && opErr.Err.Error() == "use of closed network connection" {
+			return 0, errors.New("UDPFileConn connection closed")
```

Comment:
Can we wrap the original err?
```suggestion
			return 0,  fmt.Errorf"the UDPFileConn connection closed: %w", err)
```

Reply:
We could but since we are already checking if error is equal to `use of closed network connection` 

I thought we won't be getting anything new out of error object